### PR TITLE
[MD]Update MD data source documentation link

### PR DESCRIPTION
**Pull request by kristenTian**
Signed-off-by: Kristen Tian <tyarong@amazon.com>

 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2593
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/doc_links/doc_links_service.ts [hunk 1]
```diff
@@ -403,6 +403,9 @@ export class DocLinksService {
             loadingData: `${OPENSEARCH_WEBSITE_DOCS}`,
             introduction: `${OPENSEARCH_WEBSITE_DOCS}`,
           },
+          dataSource: {
+            guide: 'https://opensearch.org/docs/latest/dashboards/discover/multi-data-sources/',
```
Comment (zhongnansu) [inline src/core/public/doc_links/doc_links_service.ts:407]: can you use `${OPENSEARCH_WEBSITE_DOCS}dashboards/discover/multi-data-sources`

Comment (kristenTian) [reply src/core/public/doc_links/doc_links_service.ts:407]: OPENSEARCH_WEBSITE_DOCS appears to have branching support.
```
    const DOC_LINK_VERSION =
      injectedMetadata.getOpenSearchDashboardsBranch() === 'main'
        ? 'latest'
        : injectedMetadata.getOpenSearchDashboardsBranch();
```
 Don't think the MD doc currently have support for this.

Comment (zhongnansu) [reply src/core/public/doc_links/doc_links_service.ts:407]: Can you confirm? I think everything on doc website is versioned, the same with MD

Comment (kristenTian) [reply src/core/public/doc_links/doc_links_service.ts]: Changed to : guide: `${OPENSEARCH_DASHBOARDS_VERSIONED_DOCS}discover/multi-data-sources/`

**Review by zhongnansu — APPROVED** [review]
lgtm

**Review by zengyan-amazon — APPROVED** [review]

**Pull request merged**